rust/src/lib/ifaces/vxlan.rs

```diff
@@ -48,18 +48,26 @@ impl VxlanInterface {
     }
 
     pub(crate) fn parent(&self) -> Option<&str> {
-        self.vxlan.as_ref().map(|cfg| cfg.base_iface.as_str())
+        self.vxlan
+            .as_ref()
+            .and_then(|cfg| cfg.base_iface.as_ref())
+            .map(|base_iface| base_iface.as_str())
     }
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Default)]
 #[serde(rename_all = "kebab-case", deny_unknown_fields)]
 #[non_exhaustive]
 pub struct VxlanConfig {
-    pub base_iface: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub base_iface: Option<String>,
     #[serde(deserialize_with = "crate::deserializer::u32_or_string")]
     pub id: u32,
     #[serde(skip_serializing_if = "Option::is_none")]
+    pub learning: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub local: Option<std::net::IpAddr>,
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
Please add `///` document here.

Reply:
Can you clarify what sort of documentation you want. Explaining what these options do in the context of VXLANs in general? Or more about the technical implementation in context of `nmstate`?

There's not much documentation for the existing fields either to align myself with

Reply:
It is crate user who look at the document like https://docs.rs/nmstate/latest/nmstate/struct.MacVlanConfig.html#structfield.accept_all_mac .

Yes, I forgot to add document for VXLAN, you may skip this.